### PR TITLE
#1758 Finance Fixes - Sort refunds newest to oldest

### LINKED ISSUE #1758: Finance Fixes - Sort refunds newest to oldest
### Description

Currently, the refunds for 'My Refunds' and 'All Refunds' are sorted from oldest to newest. They should instead be sorted from newest to oldest.

### Acceptance Criteria

- [ ] The refunds on the 'All Refunds' and 'My Refunds' tabs on the finance dashboard are sorted from newest to oldest

### PR DESCRIPTION
**Pull request by bracyw**
## Changes

Changed row to display refunds from newest to oldest (based on date + position in database).

## Notes

I had to sort by day and not time.  There was a bug where you submit {**sub1**} at 9:00 for dateA and submit {**sub2**} the next real day for dateA (but at 14:00). {**sub1**} would display first even though it was submitted before {**sub2**} because the date was the same but the time was auto generated when the refund was submitted.

Might be better to implement this elsewhere lmk.

## Test Cases

-  edge case: described above ^

## Screenshots

Max screen #1758
Min Screen #1758

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1758

### CODE REVIEW

**Comment by RChandler234** [comment]
The branch name should be longer/ a description of the change

File: src/frontend/src/pages/FinancePage/RefundsSection.tsx [hunk 1]
```diff
@@ -112,16 +112,28 @@ const Refunds = ({ userReimbursementRequests, allReimbursementRequests }: Refund
...
-                <TableRow
-                  key={`${row.date}-$${row.amount}-${index}`}
-                  sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
-                >
-                  <TableCell align="center">{datePipe(row.date)}</TableCell>
-                  <TableCell align="center">{centsToDollar(row.amount)}</TableCell>
-                  {tabValue === 1 && <TableCell align="center">{fullNamePipe(row.recipient)}</TableCell>}
-                </TableRow>
-              ))}
+              {rows
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/RefundsSection.tsx:115]: you should sort the rows above where they're defined. Also you should be able to do this in less lines. a.date is already a date so you shouldn't need to cast it as one

**Review by RChandler234 — CHANGES_REQUESTED** [review]
